### ISSUE TITLE
Feature request: Always give fortification bonus if a unit does not move. 

### ISSUE THREAD

**Issue by The-Talik** [yairm210/Unciv#12330]
### Before creating

- [X] This is NOT a gameplay feature from Civ VI, BNW, or outside - see Roadmap
- [X] This is NOT a gameplay feature from Vanilla Civ V or from G&K - If so, it should be a comment in https://github.com/yairm210/Unciv/issues/4697

### Problem Description

From an practical perspective, there is no benefit for fortifiable units to ever "wait", other than the UI annoyance of not having to find them and wake them up next turn.  Since there is no downside to fortifying, the tactical move is to always fortify, and never click wait, even if you plan on moving next round.  Right now, we have "wait" = "I want to move this unit next turn", and "forify" = "I don't want to move this unit next turn plus defensive bonus".  We should be able to choose to keep our unit in the queue, and still get the fortification bonus.

### Desired Solution

Units should be considered fortified if they meet the criteria (ie, did nothing else that round), even if you clicked wait.  This basically decouples fortification bonus, from "I want to move/view this unit next turn".

### Alternative Approaches

Alternatively, a  "fortify for 1 rnd" button could be added, but it seems less confusing just to auto apply the bonus if the criteria is met.

**Comment by sulai** [comment]
>  Right now, we have "wait" = "I want to move this unit next turn"

I agree on your idea, but the way the "wait" button currently works is that a "waiting" unit will show up again *in the same turn* when clicking the "-> next unit" button. If you click "next unit" with an idle unit selected, that one will be marked to show up next turn again. This is counter intuitive, my expectation was the same like yours. But a positive effect of the current behavior is that the "next unit" button cycles through all units only once. Maybe this is a case for a new user setting?

**Comment by RobLoach** [comment]
Fortify will fortify the unit until you stop fortifying them. The defensive bonus also isn't given immediately. It takes a few turns for it to get to 100%.

Having a unit sit there doing nothing should not provide any defensive bonus.

**Comment by itanasi** [comment]
As @Ouaz notes in #12308, this button was supposed to place the Idle Unit at the end of the queue so that you could come back to them. Next Unit should be the Skip Unit, do no action button.

**Comment by RobLoach** [comment]
I usually use the next unit button when I want to do that :upside_down_face: 


![1111](https://github.com/user-attachments/assets/c1aa7634-923d-41da-ac0a-f78c39191496)

**Comment by The-Talik** [comment]
> Fortify will fortify the unit until you stop fortifying them. The defensive bonus also isn't given immediately. It takes a few turns for it to get to 100%.
> 
> Having a unit sit there doing nothing should not provide any defensive bonus.

The case in point is that it would be nice to be able to have a unit fortify, but ALSO be in the next turn's queue so you don't forget about them.

Fortification is a game mechanic.
Sleep is a UI convenience.

From a game mechanic standpoint, there is no downside to fortifying a unit that does not move, which is why it should happen automatically.

**Comment by The-Talik** [comment]
Updated the title because this request isn't really about the behavior of the wait button.
